### PR TITLE
Add some ICEs

### PR DESCRIPTION
**Pull request by fanninpm**

### CODE REVIEW

File: ices/85447.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+impl Ctx for C {
+    type BindGroupId = u32;
+}
+
+pub struct BindGroup {
+    _id: <C as Ctx>::BindGroupId,
+}
+
+
```
Comment (JohnTitor) [inline ices/85447.rs:18]:
```suggestion
```

File: ices/85350.rs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+impl FnMut(&Context) for 'tcx {
+    fn print () -> Self :: Output{ }
+    }
+
```
Comment (JohnTitor) [inline ices/85350.rs:4]:
```suggestion
}

```

The indent looks weird.

File: ices/86053.rs [hunk 3]
```diff
@@ -0,0 +1,2 @@
+fn ordering4 < 'a , 'b     > ( a :            ,   self , self ,   self ,   self , ... ,   self ,   self , ... ) where F : FnOnce ( & 'a & 'b usize ) {
```
Comment (JohnTitor) [inline ices/86053.rs:1]: The formatting doesn't have meaning here, we can use https://github.com/rust-lang/rust/issues/86053#issuecomment-855672258 to simplify.

File: ices/86085.rs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+fn main ( ) {
+    format ! ( concat ! ( r#"lJ𐏿Æ�.𐏿�"# , "r} {}" )     ) ;
+    }
+
```
Comment (JohnTitor) [inline ices/86085.rs:4]:
```suggestion
}

```

File: ices/86086.rs [hunk 5]
```diff
@@ -0,0 +1,2 @@
+trait Tr : Sized { async extern r###"��áÖ@"### fn union < > ( # [ ']' '[' 'default' 'async' '&raw' ] & '?ဪൈ self , ... , ... , ... , ... , ... , ... , ... , ... , ... , ) -> && '_ mut ( ) ;
```
Comment (JohnTitor) [inline ices/86086.rs:1]: Note: This and some other issues are duplicates of the existing one but I'm going to include for now. We can drop them anytime.

**Review by JohnTitor — CHANGES_REQUESTED** [review]

File: ices/86053.rs (hunk 3 above)
Comment (fanninpm) [reply ices/86053.rs]: Resolved in 59274cb.

**Review by JohnTitor — APPROVED** [review]
Thanks!

**Pull request merged**